### ISSUE TITLE
Drainer error

### ISSUE THREAD

**Issue by httpcheck** [pingcap/tidb#7929]
## Question

Before asking a question, make sure you have:

- Searched existing Stack Overflow questions.
- Googled your question.
- Searched open and closed [GitHub issues](https://github.com/pingcap/tidb/issues?utf8=%E2%9C%93&q=is%3Aissue)
- Read the documentation:
  - [TiDB Readme](https://github.com/pingcap/tidb)
  - [TiDB Doc](https://github.com/pingcap/docs)

Our architecture:
![image](https://user-images.githubusercontent.com/26322785/47060589-d0d12300-d200-11e8-9863-c011e64eaf0e.png)
tidb1 : 5tidb+3pd+11tikv, total 16 machines, 128G memory + 32 cores + 900G ssd  per machine
tidb2: 3tidb+3pd+5tikv, total 8 machines, 128G memory + 32 cores + 900G ssd  per machine

I used drainer to synchronize data from the tidb1 cluster to the tidb2 cluster and mysql on June 21st, and it worked normally until August 22nd. 

![image](https://user-images.githubusercontent.com/26322785/47061158-eb0c0080-d202-11e8-921b-4e8bbc84325b.png)

I found this problem when I update the tidb2 cluster to v2.0.7. I tried to restart drainer, but it did't take effect. I even downloaded the lastest drainer.

![image](https://user-images.githubusercontent.com/26322785/47066628-4c3ece80-d219-11e8-89a5-36e0f4ef6e21.png)

**Comment by WangXiangUSTC** [comment]
It looks like drainer can't get pump's information from pd, can use sure the pd is alive?

**Comment by bianxindong** [comment]
fix: The pd nodes of tidb1 are 5.

tidb1 pd status:
![image](https://user-images.githubusercontent.com/26322785/47078151-3808ca80-d235-11e8-900c-3358cf72110d.png)

tidb1 pump's log one:
![image](https://user-images.githubusercontent.com/26322785/47078313-9e8de880-d235-11e8-84bf-4e24ff98e576.png)

tidb1 pump's log two:
![image](https://user-images.githubusercontent.com/26322785/47078364-b8c7c680-d235-11e8-9618-b792c1305494.png)

I had check the kafka cluster and it worked ok!

**Comment by WangXiangUSTC** [comment]
please use binlogctl to query pump status, use can see the usage of  binlogctl and download urls here https://github.com/pingcap/tidb-tools/tree/master/tidb_binlog/binlogctl

**Comment by bianxindong** [comment]
pump status:

![tmp6fe9aa40](https://user-images.githubusercontent.com/26322785/47125549-ce310500-d2b6-11e8-9c85-473b71c190e4.png)

**Comment by bianxindong** [comment]
There is't any error info in the log of drainer now. And I can't synchronize data from the tidb1 cluster to the tidb2 cluster. The table checkpoint of tidb2 is empty.

**Comment by WangXiangUSTC** [comment]
first please make sure you have new data insert into the source tidb. If the answer is yes, please provide me these information:
1. the pump and drainer's version, run  ./pump -v and ./drainer -v
2. please provide the goroutine information by `curl http://127.0.0.1:8249/debug/pprof/goroutine\?debug\=2 > debug.log`, use your drainer's address replace `127.0.0.1:8249`, you can send the debug.log to my email wangxiang@pingcap.com. 

thanks!

**Comment by bianxindong** [comment]
Maybe I find the reason.Just now I backuped data from another tidb cluster which had 1TB data, and the backup file was only 100GB.So I checked the log of mydumper and found the reason.

![image](https://user-images.githubusercontent.com/26322785/47141460-503f1f00-d2f3-11e8-8832-08d5b32b6598.png)
![image](https://user-images.githubusercontent.com/26322785/47141631-b4fa7980-d2f3-11e8-8ad4-6fd63d60cd7f.png)

According to your guide,I changed the variable value and tried again.

![image](https://user-images.githubusercontent.com/26322785/47141709-dd827380-d2f3-11e8-8e8a-fbb500c24de2.png)

I found the error still existed in the log of mydumper. I change the value '300m' and wait the result.
